format/spec.md

```diff
@@ -383,6 +388,13 @@ Notes:
 3. Strings are truncated to a valid UTF-8 string with no more than `L` code points.
 4. In contrast to strings, binary values do not have an assumed encoding and are truncated to `L` bytes.
 
+### XZ2 Transform Details
+
+XZ2 is based on the paper [XZ-Ordering: A Space-Filling Curve for Objects with Spatial Extensions](https://www.dbs.ifi.lmu.de/Publikationen/Boehm/Ordering_99.pdf).
```

Comment:
Dead link, can we get a reference to ArchiveX or something stable?

I found one other reference that works here, 
https://www.dbs.ifi.lmu.de/Publikationen/Papers/SSD-XZ-Order.final.pdf

But Universities are generally terrible at keeping their links up

Reply:
I think that we need clear details about how to implement this in the Iceberg spec, rather than a reference to an external paper (although that is good to include as well).

What about removing xz2 for now and adding it later? We've already made the changes to the spec to support adding new transform functions without breaking forward-compatibility, so we don't need to couple these things together. If we uncouple the additions, then it is much more likely that geometry can go into v3.